Elevate database engines in setup

ranquild opened the pull request:
Epic: https://github.com/metabase/metabase/issues/18918

Adds special UI for the most popular database engines.

How to test:
- Run a fresh Metabase instance
- During setup it should be possible to search for and select a database
- Admin database forms should be unaffected 

<img width="727" alt="Screen Shot 2021-11-11 at 8 37 43 pm" src="https://user-images.githubusercontent.com/8542534/141343723-0ef1e163-fc70-4589-9440-8d1353140afa.png">
<img width="705" alt="Screen Shot 2021-11-11 at 8 37 52 pm" src="https://user-images.githubusercontent.com/8542534/141343733-64880c34-6710-4400-88e9-efb0e1f75b0c.png">
<img width="726" alt="Screen Shot 2021-11-11 at 8 38 01 pm" src="https://user-images.githubusercontent.com/8542534/141343735-f69e39bc-4dab-4d4a-980c-602cf9af02e2.png">
<img width="702" alt="Screen Shot 2021-11-11 at 8 39 03 pm" src="https://user-images.githubusercontent.com/8542534/141345271-e12169ec-499f-4c54-ac10-fd6c070756b8.png">
<img width="714" alt="Screen Shot 2021-11-11 at 8 38 08 pm" src="https://user-images.githubusercontent.com/8542534/141343738-522b4358-c84d-4ee0-965f-a338065c7552.png">
<img width="814" alt="Screen Shot 2021-11-11 at 9 21 50 pm" src="https://user-images.githubusercontent.com/8542534/141349463-9f040816-2219-4a01-bd0f-9bd039c2945b.png">

ranquild committed:
Add engine widget

ranquild committed:
Add engine widget

ranquild committed:
Use cloud setting

ranquild committed:
Use cloud setting

ranquild committed:
Add widget

ranquild committed:
Add widget

ranquild committed:
Add missing options

ranquild committed:
Add missing options

ranquild committed:
Add missing options

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unneeded loading and error wrappers

ranquild committed:
Remove unused icon

ranquild committed:
Add svgs

ranquild committed:
Add svgs

ranquild committed:
Add extra padding for the database form

ranquild committed:
Add extra padding for the database form

ranquild committed:
Add extra padding for the database form

ranquild committed:
Add extra padding for the database form

ranquild committed:
Add extra padding for the database form

ranquild committed:
Add tests

ranquild committed:
Add tests

ranquild committed:
Add tests

ranquild committed:
Add tests

ranquild committed:
Fix tests

ranquild committed:
Add TODO

ranquild committed:
Add TODO

ranquild committed:
Add TODO

ranquild committed:
Fix lint

ranquild committed:
Fix admin case

ranquild committed:
Fix cy tests

alxnddr reviewed (changes requested):
It looks there is a names mismatch. It tries to load the following SVGs but they are added with different names: `bigquery-cloud-sdk.svg` `presto-jdbc.svg`

<img width="225" alt="Screen Shot 2021-11-15 at 13 36 17" src="https://user-images.githubusercontent.com/14301985/141758502-24911609-0847-453d-886f-85fc2cfd610f.png">

Also, not related but must be easy to fix by adding the right url. I'm wondering if you can fix this link when the presto driver is selected:
<img width="386" alt="Screen Shot 2021-11-15 at 13 39 28" src="https://user-images.githubusercontent.com/14301985/141758814-71086ae8-c243-47e6-a209-251bf75e0032.png">

ranquild committed:
Embed logo info

ranquild committed:
Fix presto driver warning

ranquild committed:
Fix presto driver warning

ranquild commented:
@alxnddr BE devs refused to add `elevated-engines` setting so I've moved the array to `lib/engines.js`. I've also fixed icons and link issues, please have a look.

alxnddr reviewed (approved):
👍

kulyk commented on frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.jsx:
```diff
@@ -0,0 +1,201 @@
+    <EngineEmptyStateRoot>
+      <EngineEmptyIcon name="search" size={32} />
+      <EngineEmptyText>{jt`Don’t see your database? Check out our ${(
+        <ExternalLink href="https://www.metabase.com/docs/latest/developers-guide-drivers.html">
+          {t`Community Drivers`}
+        </ExternalLink>
```
I suppose we should hide the Community Drivers link for cloud instances. AFAIK, community drivers are not currently supported there (I can be wrong though)

ranquild replied:
Good catch!

ranquild committed:
Merge remote-tracking branch 'origin/master' into 18918-elevate-dbs

# Conflicts:
#	frontend/src/metabase/entities/databases/forms.js

ranquild committed:
Fix merge conflicts

ranquild committed:
Add cloud case
